platforms/ios/lib/WysiwygComposer/Sources/HTMLParser/Extensions/UITextView.swift

```diff
@@ -25,14 +25,33 @@ public extension UITextView {
             .compactMap { $0 as? BackgroundStyleLayer }
             .forEach { $0.removeFromSuperlayer() }
 
-        attributedText.enumerateTypedAttribute(.backgroundStyle) { (style: BackgroundStyle, range: NSRange, _) in
-            let glyphRange = layoutManager.glyphRange(forCharacterRange: range, actualCharacterRange: nil)
-            let rect = layoutManager
-                .boundingRect(forGlyphRange: glyphRange, in: self.textContainer)
-                .extendHorizontally(in: frame)
-
-            let styleLayer = BackgroundStyleLayer(style: style, frame: rect)
+        attributedText.enumerateTypedAttribute(.blockStyle) { (style: BlockStyle, range: NSRange, _) in
+            var styleLayer: BackgroundStyleLayer?
+            switch style.type {
+            case .background:
+                let glyphRange = layoutManager.glyphRange(forCharacterRange: range, actualCharacterRange: nil)
+                let rect = layoutManager
+                    .boundingRect(forGlyphRange: glyphRange, in: self.textContainer)
+                    .extendHorizontally(in: frame)
 
+                styleLayer = BackgroundStyleLayer(style: style, frame: rect)
+            case .side:
+                let beginning = self.beginningOfDocument
+                guard let start = self.position(from: beginning, offset: range.location) else { return }
+                // removing the last character to avoid including the empty new line
+                guard let end = self.position(from: start, offset: range.length - 1) else { return }
+                guard let range = self.textRange(from: start, to: end) else { return }
+                let rects = selectionRects(for: range)
+                var textRect = CGRect.null
+                for rect in rects {
+                    textRect = textRect.union(rect.rect)
+                }
+                if !textRect.isNull {
+                    let rect = CGRect(x: 5, y: textRect.origin.y, width: 4, height: textRect.size.height)
+                    styleLayer = BackgroundStyleLayer(style: style, frame: rect)
+                }
```

Comment:
```suggestion
                let glyphRange = layoutManager.glyphRange(forCharacterRange: range, actualCharacterRange: nil)
                let textRect = layoutManager
                    .boundingRect(forGlyphRange: glyphRange, in: self.textContainer)
                let rect = CGRect(x: 5, y: textRect.origin.y, width: 4, height: textRect.size.height)
                styleLayer = BackgroundStyleLayer(style: style, frame: rect)
```

Reply:
I think that the indent issue is not related on this but  actually on the paragraph adding that we do in `applyBackgroundStyles(style: HTMLParserStyle)`

Reply:
Oh yeah it's definitely not related to this piece of code. Basically we'll have either `\n(ZWSP)\n` or `(Some text)\n` at the end of code block/quotes. 
The very last `\n` shouldn't have the code block / quote paragraph style, but the standard paragraph style instead (as this represents the newline between the block and the next line)

Reply:
Tried to fix the issue by doing this when adding the paragraph style but... doesn't seem to work?
```
 if self.character(at: range.location + range.length - 1) == "\n" {
                    addAttribute(.paragraphStyle, value: NSParagraphStyle.default, range: NSRange(location: range.location + range.length - 1, length: 1))
                }
```